packages/lodestar/src/network/gossip/gossipsub.ts

```diff
@@ -70,16 +69,22 @@ export class LodestarGossipsub extends Gossipsub {
     }
   }
 
-  public async validate(rawMessage: Message, peer: Peer): Promise<boolean> {
-    const message: ILodestarGossipMessage = normalizeInRpcMessage(rawMessage);
+  /**
+   * Override https://github.com/ChainSafe/js-libp2p-gossipsub/blob/v0.6.0/ts/index.ts#L442
+   * Should throw ERR_TOPIC_VALIDATOR_IGNORE or ERR_TOPIC_VALIDATOR_REJECT
+   * If we don't throw it means accept
+   * @param rawMessage
+   */
+  public async validate(rawMessage: InMessage): Promise<void> {
```

Comment:
We need to rely on super.validate, and call it bc we track rejected / ignored messages in super.validate.
One direction might be to push all validation into the topic validators.

```typescript

async function genericIsValid (message: InMessage): Promise<boolean> {
  return (
    message.topicIDs && message.topicIDs.length === 1 &&
    message.data && message.data.length > GOSSIP_MAX_SIZE
  );
}

async function topicValidator (topic: string, objValidatorFn: (obj: GossipObject) => Promise<void>): (msg: InMessage) => Promise<void> {
  return async (topic: string, msg: InMessage): Promise<void> => {
    if (!genericIsValid(msg)) {
      throw new GossipValidationError(ERR_TOPIC_VALIDATOR_REJECT);
    }
    try {
      const {object} = deserializeGossipMessage(topic, message);
      await objValidatorFn(object);
    } catch (e) {
      throw new GossipValidationError(ERR_TOPIC_VALIDATOR_REJECT);
    }
  }
}

// in gossip.ts / gossipsub.ts, set the topic validators
// they fail fast here: https://github.com/libp2p/js-libp2p-interfaces/blob/master/src/pubsub/index.js#L515
this.topicValidators.set(
  getGossipTopic(BLOCK),
  topicValidator(getGossipTopic(BLOCK), validateGossipBlock)
);
```
Then we wouldn't override `validate` at all, just rely on the underlying implementation.
This could be cleaned up, just showing the idea.